### PR TITLE
Add E2E tests to cover Publish dropdown options

### LINKED ISSUE #46593: [E2E test coverage]: Publish dropdown options
Add E2E tests to cover the Publish dropdown options.

This will test if it:

- can copy a product to a new draft.
- can schedule a product publication
- can move the product to the trash can

### PR DESCRIPTION
**Pull request by octaedro**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR adds tests to cover the Publish dropdown options. It will test the `Schedule` a product publication, the `Copy to new draft` option and the `Move to trash` functionallity.   

Screenshot 2024-04-17 at 1 07 56 PM

Closes #46593.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout this branch
2. Run `cd plugins/woocommerce && pnpm env:start`
3. Then run `USE_WP_ENV=1 pnpm playwright test --config=tests/e2e-pw/playwright.config.js` if you add the relative path behind it will run the file you want. I would also add `--headed` to see visually what's going on. In this case, the command should look like `USE_WP_ENV=1 pnpm playwright test --config=tests/e2e-pw/playwright.config.js ./tests/e2e-pw/tests/merchant/products/block-editor/product-edit-block-editor.spec.js --headed`
4. All the tests should pass.

More details [here](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/tests/e2e-pw/README.md).

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/product-edit-block-editor.spec.js [hunk 1]
```diff
@@ -80,3 +80,70 @@ test( 'can update the general information of a product', async ( {
 			.toHaveText( updatedProduct.description );
 	} );
 } );
+
+test( 'can schedule a product publication', async ( { page, product } ) => {
```
Comment (louwie17) [inline plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/product-edit-block-editor.spec.js]: This test schedules a product, duplicates a product and removes a product.
Should we not move these into individual tests?

**Review by louwie17 — CHANGES_REQUESTED** [review]
Thanks for adding this Fernando, the E2E test ran fine locally, I just think we should split it up in a couple different tests if possible?

Also in relation to your testing instructions:

> Run our E2E tests. All the tests should pass.

It be great if you could be more specific here in the future, for example: list the commands that are required for E2E tests. Not everyone on the team will be aware of how to do this and it would save the team a lot of time if they don't need to go looking for it. Linking to the E2E docs would also help.

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/product-edit-block-editor.spec.js (hunk 1 above)
Comment (octaedro) [reply plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/product-edit-block-editor.spec.js]: Good idea! Done in https://github.com/woocommerce/woocommerce/pull/46658/commits/03ac99954746babfc0b1459b4bd7c0d1c61694e4

**Comment by octaedro** [comment]
Thank you @louwie17 for your review.

I updated the testing instructions and addressed the suggestion you made.
Could you take another look at this PR?

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/product-edit-block-editor.spec.js [hunk 2]
```diff
@@ -80,3 +80,84 @@ test( 'can update the general information of a product', async ( {
...
+			.getByText( '14' )
+			.click();
+
+		await page.getByRole( 'button', { name: 'Schedule' } ).click();
+
+		await expect(
+			page.getByLabel( 'Dismiss this notice' ).first()
+		).toContainText( 'Product scheduled for' );
+	} );
+	test( 'can dupicate a product', async ( { page, product } ) => {
```
Comment (louwie17) [inline plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/product-edit-block-editor.spec.js]:
```suggestion
	test( 'can duplicate a product', async ( { page, product } ) => {
```
Small spelling mistake

**Review by louwie17 — APPROVED** [review]
LGTM 🚀 Thanks for the updates @octaedro, just a small spelling mistake left but otherwise all is good.

**Pull request merged**